src/h3lib/lib/vertex.c

```diff
@@ -132,3 +137,136 @@ int vertexNumForDirection(const H3Index origin, const Direction direction) {
+
+    // Find the appropriate direction, rotating CW if necessary
+    return isPentagon ? vertexNumToDirectionPent[(vertexNum + rotations) %
+                                                 NUM_PENT_VERTS]
+                      : vertexNumToDirectionHex[(vertexNum + rotations) %
+                                                NUM_HEX_VERTS];
+}
+
+/**
+ * Get a single vertex for a given cell, as an H3 index, or
+ * H3_NULL if the vertex is invalid
+ * @param cell    Cell to get the vertex for
+ * @param vertexNum Number (index) of the vertex to calculate
+ */
+H3Index H3_EXPORT(cellToVertex)(H3Index cell, int vertexNum) {
+    int cellIsPentagon = H3_EXPORT(h3IsPentagon)(cell);
+    int cellNumVerts = cellIsPentagon ? NUM_PENT_VERTS : NUM_HEX_VERTS;
+
+    // Get the left neighbor of the vertex, with its rotations
+    Direction left = directionForVertexNum(cell, vertexNum);
+    if (left == INVALID_DIGIT) return H3_NULL;
+    int rRotations = 0;
+    H3Index leftNeighbor = h3NeighborRotations(cell, left, &rRotations);
+
+    // Get the right neighbor of the vertex, with its rotations
+    // Note that vertex - 1 is the right side, as vertex numbers are CCW
+    Direction right = directionForVertexNum(
+        cell, (vertexNum - 1 + cellNumVerts) % cellNumVerts);
+    // This case should be unreachable; invalid verts  fail the left side first
+    if (right == INVALID_DIGIT) return H3_NULL;  // LCOV_EXCL_LINE
+    int lRotations = 0;
+    H3Index rightNeighbor = h3NeighborRotations(cell, right, &lRotations);
+
+    // Determine the owner. By convention, this is the cell with the
+    // lowest numerical index.
+    H3Index owner = cell;
+    if (leftNeighbor < owner) owner = leftNeighbor;
+    if (rightNeighbor < owner) owner = rightNeighbor;
+
+    // Determine the vertex number for the owner cell
+    int ownerVertexNum = vertexNum;
+
+    if (owner == leftNeighbor) {
+        Direction dir = directionForNeighbor(owner, cell);
+        // For the left neighbor, we need the second vertex of the
+        // edge, which may involve looping around the vertex nums
+        ownerVertexNum = vertexNumForDirection(owner, dir) + 1;
+        if (ownerVertexNum == NUM_HEX_VERTS ||
+            (H3_EXPORT(h3IsPentagon)(owner) &&
+             ownerVertexNum == NUM_PENT_VERTS)) {
+            ownerVertexNum = 0;
+        }
+        // }
+    } else if (owner == rightNeighbor) {
+        Direction dir = directionForNeighbor(owner, cell);
+        ownerVertexNum = vertexNumForDirection(owner, dir);
+    }
+
+    // Create the vertex index
+    H3Index vertex = owner;
+    H3_SET_MODE(vertex, H3_VERTEX_MODE);
+    H3_SET_RESERVED_BITS(vertex, ownerVertexNum);
+
+    return vertex;
+}
+
+/**
+ * Get all vertexes for the given cell
+ * @param cell      Cell to get the vertexes for
+ * @param vertexes  Array to hold vertex output. Must have length >= 6.
+ */
+void H3_EXPORT(cellToVertexes)(H3Index cell, H3Index* vertexes) {
```

Comment:
as part of the return code work I imagine these function signatures will get updated. Can't do that in this PR because some needed types are not present.

Reply:
Possibly? There isn't an error case for this function if `cell` is valid (and, per similar discussions, I'm not verifying that here). I'm actually not certain what would happen here if an invalid cell is passed in (likely the output is undefined, possibly it's all `H3_NULL`)

Reply:
As with #359 I'd like to make all public functions have a mechanism to return error information since we don't have an exception capacity otherwise.

Reply:
Right, that makes sense. So we'd include the mechanism even when unused.